### PR TITLE
Yac: enable eternal voting

### PR DESCRIPTION
**Pull request by MBoldyrev**
Signed-off-by: Mikhail Boldyrev <miboldyrev@gmail.com>

### Description of the Change

make Yac send votes through all peers in the order until the round closes.

### Benefits

hopefully solves some issues during startup & networking problems.

### CODE REVIEW

File: test/module/irohad/consensus/yac/yac_simple_cold_case_test.cpp [hunk 1]
```diff
@@ -27,18 +28,18 @@ using namespace framework::test_subscriber;
...
-  cout << "----------|YacWhenAchieveOneVote|----------" << endl;
-
-  EXPECT_CALL(*network, sendState(_, _)).Times(default_peers.size());
-
   YacHash my_hash(initial_round, "my_proposal_hash", "my_block_hash");
 
   auto order = ClusterOrdering::create(default_peers);
   ASSERT_TRUE(order);
 
+  setNetworkOrderCheckerSingleVote(order.value(), my_hash, 20);
```
Comment (lebdron) [inline test/module/irohad/consensus/yac/yac_simple_cold_case_test.cpp:41]: Maybe move 20 to fixture? Same below

File: test/module/irohad/consensus/yac/yac_sunny_day_test.cpp [hunk 2]
```diff
@@ -68,12 +74,12 @@ TEST_F(YacTest, ValidCaseWhenReceiveCommit) {
     ASSERT_EQ(my_hash, boost::get<CommitMessage>(val).votes.at(0).hash);
   });
 
-  EXPECT_CALL(*network, sendState(_, _)).Times(my_peers.size());
-
   EXPECT_CALL(*timer, deny()).Times(AtLeast(1));
 
   EXPECT_CALL(*crypto, verify(_)).WillRepeatedly(Return(true));
 
+  setNetworkOrderCheckerSingleVote(my_order.value(), my_hash, 42);
```
Comment (lebdron) [inline test/module/irohad/consensus/yac/yac_sunny_day_test.cpp:81]: Same as above for constant in fixture

File: test/module/irohad/consensus/yac/yac_sunny_day_test.cpp [hunk 3]
```diff
@@ -156,9 +160,17 @@ TEST_F(YacTest, ValidCaseWhenSoloConsensus) {
     ASSERT_EQ(my_hash, boost::get<CommitMessage>(val).votes.at(0).hash);
   });
 
-  yac->vote(my_hash, my_order.value());
-
   auto vote_message = createVote(my_hash, std::to_string(0));
+  std::vector<VoteMessage> votes{{vote_message}};
```
Comment (lebdron) [inline test/module/irohad/consensus/yac/yac_sunny_day_test.cpp:164]:
```suggestion
```

File: test/module/irohad/consensus/yac/yac_synchronization_test.cpp [hunk 4]
```diff
@@ -25,29 +35,23 @@ class NetworkUtil {
...
+
   /// create votes of peers by their number
-  auto createVotes(
-      const std::vector<size_t> &peers,
-      const iroha::consensus::Round &r,
-      const std::string &block_hash = "default_block",
-      const std::string &proposal_hash = "default_proposal") const {
+  /// @param peers indices of peers in @a peers_
+  /// @param hash for all votes
+  /// @return vactor of votes for the @a hash from each of @peers
```
Comment (lebdron) [inline test/module/irohad/consensus/yac/yac_synchronization_test.cpp:49]:
```suggestion
  /// @return vector of votes for the @a hash from each of @peers
```

File: test/module/irohad/consensus/yac/yac_fixture.hpp [hunk 5]
```diff
@@ -77,6 +77,102 @@ namespace iroha {
...
+        /**
+         * Make a checker of sendState invocations that matches the destination
+         * peer with the @a order.
+         * @param order the order to check
+         * @return a lambda to be executed in sendState mock function.
+         */
+        auto makeSendStateOrderChecker(const ClusterOrdering &order) {
+          auto times_sent_state = std::make_shared<size_t>(0);
+          return [&peers = order.getPeers(), times_sent_state](
+                     const auto &peer, const auto & /* state */) {
```
[the remark this answers is not in the document]
Comment (MBoldyrev) [reply test/module/irohad/consensus/yac/yac_fixture.hpp:91]: it doesn't work that way =(

**Pull request merged**